src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/conditions/ExclusionZoneService.kt

```diff
@@ -82,66 +80,85 @@ class ExclusionZoneService(
+    val uploadDetail = additionalConditionUploadDetail(
+      additionalCondition(conditionId).additionalConditionUploadSummary.first().uploadDetailId,
+    )
 
-      return Triple(originalDataDsUuid, fullSizeImageDsUuid, thumbnailImageDsUuid)
+    return if (uploadDetail.fullSizeImageDsUuid != null) {
+      documentService.downloadDocument(UUID.fromString(uploadDetail.fullSizeImageDsUuid))
+    } else {
+      uploadDetail.fullSizeImage
     }
   }
 
-  private fun removeExistingUploads(additionalCondition: AdditionalCondition) {
-    additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }.forEach {
-      additionalConditionUploadDetailRepository.findById(it).ifPresent { detail ->
-        additionalConditionUploadDetailRepository.delete(detail)
-      }
-    }
+  fun preloadThumbnailsFor(licence: Licence) {
+    licence.additionalConditions
+      .flatMap { it.additionalConditionUploadSummary }
+      .filterNot { it.thumbnailImageDsUuid == null }
+      .forEach { it.preloadedThumbnailImage = documentService.downloadDocument(UUID.fromString(it.thumbnailImageDsUuid)) }
   }
 
-  fun getExclusionZoneImage(licenceId: Long, conditionId: Long): ByteArray? {
-    licenceRepository
-      .findById(licenceId)
-      .orElseThrow { EntityNotFoundException("$licenceId") }
+  fun deleteDocumentsFor(licence: Licence) {
+    log.info("Deleting documents for Licence id=${licence.id}")
 
-    val additionalCondition = additionalConditionRepository
-      .findById(conditionId)
-      .orElseThrow { EntityNotFoundException("$conditionId") }
+    deleteDocumentsFor(licence.additionalConditions)
+  }
 
-    val uploadIds = additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }
-    if (uploadIds.isEmpty()) {
-      throw EntityNotFoundException("$conditionId")
-    }
+  fun deleteDocumentsFor(additionalConditions: List<AdditionalCondition>) {
+    log.info("Deleting documents for ${additionalConditions.size} additional conditions")
 
-    val upload = additionalConditionUploadDetailRepository
-      .findById(uploadIds.first())
-      .orElseThrow { EntityNotFoundException("$conditionId") }
+    val uploadSummaries = additionalConditions.flatMap { it.additionalConditionUploadSummary }
+    val uuidsToDelete = mutableListOf<String?>()
 
-    return if (upload.fullSizeImageDsUuid != null) {
-      documentService.downloadDocument(UUID.fromString(upload.fullSizeImageDsUuid))
-    } else {
-      upload.fullSizeImage
-    }
+    // Remove AdditionalConditionUploadSummary documents
+    uploadSummaries.forEach { uuidsToDelete.add(it.thumbnailImageDsUuid) }
+
+    // Remove AdditionalConditionUploadDetail documents
+    uploadSummaries
+      .map { additionalConditionUploadDetailRepository.findById(it.uploadDetailId).orElseThrow() }
+      .forEach { uuidsToDelete.addAll(listOf(it.fullSizeImageDsUuid, it.originalDataDsUuid)) }
+
+    uuidsToDelete.filterNotNull()
```

Comment:
we need to make sure that the document UUIDs aren't still being used by another existing licence.
 
e.g:
* Create a licence with exclusion zone
* Vary the licence, creates a copy
* remove the exclusion zone condition from the variation
* exclusion zone still visible on the original licence pdf

Reply:
gotcha, I will take a look at this now

Reply:
The solution I have opted for is to count the number of times a specific UUID exists in the DB globally and if it appears only once then it is safe to delete. See DocumentCountsRepository.